pkg/receive/handler.go

```diff
@@ -103,35 +79,20 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 		),
 	}
 
-	router := route.New().WithInstrumentation(h.instrumentHandler)
-	h.router = router
+	ins := extpromhttp.NewNopInstrumentationMiddleware()
+	if o.Registry != nil {
+		ins = extpromhttp.NewInstrumentationMiddleware(o.Registry)
+		o.Registry.MustRegister(h.forwardRequestsTotal)
```

Comment:
Did we always forget to register this or am I missing why this line is here?

Reply:
If I understood you correctly, It was registered before. I just removed the ones which come from `InstrumentationMiddleware`.
https://github.com/thanos-io/thanos/blob/8feb0798a68e2a11b9bd625fcc80842f0d04a758/pkg/receive/handler.go#L112-L119

Reply:
Got it. Makes sense now.